CIP-0035/README.md

```diff
@@ -0,0 +1,52 @@
+---
+CIP: 35
+Title: Daedalus URI Scheme
+Authors: Daniel Main <daniel.main@iohk.io>
+Comments-URI:
+- https://github.com/cardano-foundation/CIPs/pull/130
+Status: Draft
+Type: Informational
+Created: 2022-03-18
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This proposal is an extension of CIP-0013, which describes a basic URI scheme to handle ada transfers or partial transactions with Daedalus or any desktop Wallet installed in the operating system.
+
+# 💡 Motivation
+
+DEX, Web-dApps and Desktop-dApps (e.g. games or desktop applications) should have the possibility to interact directly with desktop wallets. By defining a URI scheme and registering it in the operating system, it will allow opening Daedalus or any desktop wallet same as when you call or click a mailto:email@sample.com and your email client is opened.
+
+An application could ask for a payment and offer options like PayPal, Visa, cardano-web. Although a user might want to help cardano to get more decentralized using a full-node-wallet that is why the application could also offer payments with a cardano-desktop.
+The usage of a different URL will solve the problem of users, which have multiple wallets: browser-wallet for daily usage and a secure desktop wallet for the savings.
+
+# 📖 Specification
+
+The core implementation should follow the [BIP-21](https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki) standard (with `bitcoin:` replaced with `desktop+cardano:`)
+
+Correct examples:
+```
+<a href="desktop+cardano-mainnet://Ae2tdPwUPEZ76BjmWDTS7poTekAvNqBjgfthF92pSLSDVpRVnLP7meaFhVd">Buy/Send/Donate</a>
```

Comment:
Why is the URI scheme prefix changed from `web+cardano` in https://github.com/cardano-foundation/CIPs/pull/130 to `desktop+cardano` in this PR?  CIP-0013 itself specifies `web+cardano` which as far as I know dates back to https://github.com/Emurgo/yoroi-frontend/pull/511.

`The web+` prefix in turn was chosen because simpler scheme literals proposed earlier like `cardano` and `ada` (https://github.com/input-output-hk/daedalus/issues/883) would have [required safelisting with IANA](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax) while anything beginning with `web+` could be freely used on the fly (i.e. by apps registering `web+cardano` as a handler).  It doesn't look like `desktop+` has this same property: https://html.spec.whatwg.org/multipage/system-state.html#custom-handlers

Reply:
Thank you @rphair for the review!

We need a distinction because there will be cases of users, which will have multiple wallets installed, secure ones like Daedalus (full-node) for savings and an unsecure one (any light-wallet) for fast transactions.

What would happen by clicking on a `web+cardano`? Which wallet will be opened desktop or web is the dilema 🤷‍♂? 

> It doesn't look like desktop+ has this same property: [html.spec.whatwg.org/multipage/system-state.html#custom-handlers](https://html.spec.whatwg.org/multipage/system-state.html#custom-handlers)

I see the point that it needs to start with `web+` but how to distinguish them between desktop and light-wallets?

`web+desktop+cardano` ?

Also why other apps like visual studio code can be opened with `vscode://` 🤔
https://code.visualstudio.com/docs/editor/command-line#_opening-vs-code-with-urls

Reply:
@danielmain the reason they can use `vscode://` is because the restriction for `web+` does not apply to desktop apps.

> how to distinguish them between desktop and light-wallets

I don't see how this requirement matters. It's entirely possible somebody publishes a desktop application that works as a light wallet and it's entirely possible that somebody has multiple fullnode wallets installed in the future if somebody ever creates another fullnode wallet for Cardano. Prompting the user about which app they want to use for a URI scheme is something handled at the OS level for this reason. For example, if you have multiple mail clients installed, clicking on foo@foo.com will prompt you asking you which mail client you want to use for the URL.

Reply:
> @danielmain the reason they can use vscode:// is because the restriction for web+ does not apply to desktop apps.

Well, Daedalus is, in fact, a desktop app 😅 that is why I did not understand the reason to use `web+`

Anyway, I could live with the idea to use only one protocol for any wallet (not so friendly for users using two wallets and switching between light/desktop in the OS settings). Just we would need to point to mainnet/flight/testnet and add an extra parameter with the unix-timestamp plus a timeout.

I will update this PR.

thank you for your feedback!